module/VuFind/src/VuFind/Autocomplete/Solr.php

```diff
@@ -360,6 +376,18 @@ protected function setSortField($new)
         $this->sortField = $new;
     }
 
+    /**
+     * Set the limit. Useful for child classes.
+     *
+     * @param int $new Limit of search result rows.
+     *
+     * @return void
+     */
+    protected function setLimit($new)
```

Comment:
Might as well add types here:

```suggestion
    protected function setLimit(int $new): void
```

I'd also suggest that maybe `$limit` would be a more meaningful parameter name than `$new`. Also, an open question: should we make the type `?int` and adjust the comment to note that setting this to null will use whatever the default limit of the search backend is?

Reply:
The reason I choose $new were the methods `setDisplayField()` and `setSortField()` which also use $new. I would have named it $limit as well but decided to be consistent with those methods. Making it of type `?int` is better and setting it to null defaults to the search backend. We may as well consider to check for values > 0 and setting it to null otherwise.

Reply:
It makes sense that you followed precedent, but I'd say you can feel free to use the better name. This class was originally written a long time ago, and I don't think my past choices were the best there. :-) Your suggested validation adjustment also makes sense!